### PR TITLE
Add Shandong network and fork

### PR DESCRIPTION
**Pull request by shemnon**
## PR description

Add the transient Shandong fork and network definitions.  For Shanghai fork testing. Besu doesn't sync, but it does connect.

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if
    [updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).

## Changelog

- [X] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

**Comment by shemnon** [comment]
There's discussion of resetting Shandong with a new chain ID again on Saturday 5 Nov, so this won't merge until next week.

File: config/src/main/resources/shandong.json [hunk 1]
```diff
@@ -0,0 +1,870 @@
+{
+  "config": {
+    "chainId": 1337902,
+    "londonBlock": 0,
```
Comment (shemnon) [inline config/src/main/resources/shandong.json:4]: Due to needing basefee in the genesis we add londonblock zero to trigger the logic.  In a future PR we should revisit this.

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/ProtocolScheduleBuilder.java [hunk 2]
```diff
@@ -211,54 +210,46 @@ public ProtocolSchedule createProtocolSchedule() {
...
+            create(config.getEcip1049BlockNumber(), specFactory.ecip1049Definition()))
+        .stream()
+        .filter(Optional::isPresent)
+        .map(Optional::get)
+        .collect(
+            Collectors.toMap(
+                BuilderMapEntry::getBlock,
+                b -> b,
+                (existing, replacement) -> replacement,
+                TreeMap::new));
```
Comment (diega) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/ProtocolScheduleBuilder.java:252]: If we are changing this snippet I suggest to remove the usage of `c.g.c.c.Lists#newArrayList` too
```suggestion
    final TreeMap<Long, BuilderMapEntry> builders = new TreeMap<>();
    create(OptionalLong.of(0), specFactory.frontierDefinition())
        .ifPresent(b -> builders.put(b.getBlock(), b));
    create(config.getHomesteadBlockNumber(), specFactory.homesteadDefinition())
        .ifPresent(b -> builders.put(b.getBlock(), b));
    create(config.getTangerineWhistleBlockNumber(), specFactory.tangerineWhistleDefinition())
        .ifPresent(b -> builders.put(b.getBlock(), b));
    create(config.getSpuriousDragonBlockNumber(), specFactory.spuriousDragonDefinition())
        .ifPresent(b -> builders.put(b.getBlock(), b));
    create(config.getByzantiumBlockNumber(), specFactory.byzantiumDefinition())
        .ifPresent(b -> builders.put(b.getBlock(), b));
    create(config.getConstantinopleBlockNumber(), specFactory.constantinopleDefinition())
        .ifPresent(b -> builders.put(b.getBlock(), b));
    create(config.getPetersburgBlockNumber(), specFactory.petersburgDefinition())
        .ifPresent(b -> builders.put(b.getBlock(), b));
    create(config.getIstanbulBlockNumber(), specFactory.istanbulDefinition())
        .ifPresent(b -> builders.put(b.getBlock(), b));
    create(config.getMuirGlacierBlockNumber(), specFactory.muirGlacierDefinition())
        .ifPresent(b -> builders.put(b.getBlock(), b));
    create(config.getBerlinBlockNumber(), specFactory.berlinDefinition())
        .ifPresent(b -> builders.put(b.getBlock(), b));
    create(config.getLondonBlockNumber(), specFactory.londonDefinition(config))
        .ifPresent(b -> builders.put(b.getBlock(), b));
    create(config.getArrowGlacierBlockNumber(), specFactory.arrowGlacierDefinition(config))
        .ifPresent(b -> builders.put(b.getBlock(), b));
    create(config.getGrayGlacierBlockNumber(), specFactory.grayGlacierDefinition(config))
        .ifPresent(b -> builders.put(b.getBlock(), b));
    create(config.getMergeNetSplitBlockNumber(), specFactory.parisDefinition(config))
        .ifPresent(b -> builders.put(b.getBlock(), b));
    // Classic Milestones
    create(config.getEcip1015BlockNumber(), specFactory.tangerineWhistleDefinition())
        .ifPresent(b -> builders.put(b.getBlock(), b));
    create(config.getDieHardBlockNumber(), specFactory.dieHardDefinition())
        .ifPresent(b -> builders.put(b.getBlock(), b));
    create(config.getGothamBlockNumber(), specFactory.gothamDefinition())
        .ifPresent(b -> builders.put(b.getBlock(), b));
    create(
            config.getDefuseDifficultyBombBlockNumber(),
            specFactory.defuseDifficultyBombDefinition())
        .ifPresent(b -> builders.put(b.getBlock(), b));
    create(config.getAtlantisBlockNumber(), specFactory.atlantisDefinition())
        .ifPresent(b -> builders.put(b.getBlock(), b));
    create(config.getAghartaBlockNumber(), specFactory.aghartaDefinition())
        .ifPresent(b -> builders.put(b.getBlock(), b));
    create(config.getPhoenixBlockNumber(), specFactory.phoenixDefinition())
        .ifPresent(b -> builders.put(b.getBlock(), b));
    create(config.getThanosBlockNumber(), specFactory.thanosDefinition())
        .ifPresent(b -> builders.put(b.getBlock(), b));
    create(config.getMagnetoBlockNumber(), specFactory.magnetoDefinition())
        .ifPresent(b -> builders.put(b.getBlock(), b));
    create(config.getMystiqueBlockNumber(), specFactory.mystiqueDefinition())
        .ifPresent(b -> builders.put(b.getBlock(), b));
    create(config.getEcip1049BlockNumber(), specFactory.ecip1049Definition())
        .ifPresent(b -> builders.put(b.getBlock(), b));
    return builders;

```

**Review by diega — APPROVED** [review]
LGTM!

File: config/src/main/resources/shandong.json (hunk 1 above)
Comment (shemnon) [reply config/src/main/resources/shandong.json:4]: We do extend the [parisDefinition](https://github.com/hyperledger/besu/blob/c4c0846976baae6b5fac9637a35e4f9b817bdec3/ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/MainnetProtocolSpecs.java#L642).

It's [this check](https://github.com/hyperledger/besu/blob/c4c0846976baae6b5fac9637a35e4f9b817bdec3/config/src/main/java/org/hyperledger/besu/config/GenesisConfigFile.java#L147) that requires the setting of any value for LondonBlock. I'll fix it in a separate PR.

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/ProtocolScheduleBuilder.java (hunk 2 above)
Comment (shemnon) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/ProtocolScheduleBuilder.java]: Done, but slightly differently.  Using `Stream.of`

**Pull request merged**